src/main/scripts/preflight.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+#      Copyright (c) Microsoft Corporation.
```

Comment:
Same story here.  Why not `set -Eeuo pipefail`?

Reply:
We do parameters validation in this script and will output explicit and meaningful error message if any validation failed, which is helpful for user to trouble shoot the issue. So we want to control when to exit successfully or with failure by ourselves, instead of being controlled by these flags at all.